nsxt/resource_nsxt_policy_host_transport_node_collection.go

```diff
@@ -267,6 +277,14 @@ func resourceNsxtPolicyHostTransportNodeCollectionDelete(d *schema.ResourceData,
 		return err
 	}
 
+	removeNsxOnDestroy := d.Get("remove_nsx_on_destroy").(bool)
+	if removeNsxOnDestroy {
+		log.Printf("[INFO] Removing NSX from hosts associated with HostTransportNodeCollection with ID %s", id)
+		err = client.Removensx(siteID, epID, id)
```

Comment:
Is this API call synchronous or asynchronous?
I seem to recall the latter.

I'm trying to look at the problem from a different perspective.
I wonder if this attribute should set an expectation in the user that NSX will be uninstalled from nodes.
In that case we might need to find a way to confirm deletion was done successfully; if not fail the TF apply operation, and the TN collection should be kept in the state.

Does this make sense?

Reply:
This call is asynchronous. However, upon calling this API, the host transport node collection state becomes unavailable.
We could busy-wait on the associated compute collection status until removal is complete. Does that make sense?